roachprod: flags, start options cleanup

RaduBerinde opened the pull request:
#### roachprod: extract flags and bash completion code

This change moves the flags and bash completion code to make the code
easier to navigate.

Release note: None

#### roachprod: minor flags cleanup

This change:
 - makes the formatting consistent when setting up flags;
 - reorganizes and simplifies the code that sets up flags for multiple
   commands at a time;
 - removes global values that are redundant.

Release note: None

#### roachprod: move DebugDir to ClusterSettings


#### roachprod: clean up Nodes initialization

This moves the initialization of Nodes into `NewSyncedCluster`, rather
than expecting the caller to modify the public field directly.

Release note: None

#### roachprod: clean up start options

This commit removes the `install.StartOpts` global and instead plumbs
it from `Start()`. The extra start args are also moved into this
structure.

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.png" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/72826)
<!-- Reviewable:end -->

RaduBerinde committed:
roachprod: extract flags and bash completion code

This change moves the flags and bash completion code to make the code
easier to navigate.

Release note: None

RaduBerinde committed:
roachprod: minor flags cleanup

This change:
 - makes the formatting consistent when setting up flags;
 - reorganizes and simplifies the code that sets up flags for multiple
   commands at a time;
 - removes global values that are redundant.

Release note: None

RaduBerinde committed:
roachprod: clean up start options

This commit removes the `install.StartOpts` global and instead plumbs
it from `Start()`. The extra start args are also moved into this
structure.

Release note: None

RaduBerinde committed:
roachprod: move DebugDir to ClusterSettings

RaduBerinde committed:
roachprod: clean up Nodes initialization

This moves the initialization of Nodes into `NewSyncedCluster`, rather
than expecting the caller to modify the public field directly.

Release note: None

rail reviewed (approved):
Thank you for the clean up!

> Reviewed 4 of 4 files at r1, 1 of 1 files at r2, 2 of 2 files at r3, 2 of 2 files at r4, 5 of 5 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/72826#-:-MoiXq91-2gzu5ddsb90:b-4hp8hf)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @healthy-pod and @stevendanna)



<!-- Sent from Reviewable.io -->

RaduBerinde commented:
TFTR!

bors r+
